src/app/app.component.ts

```diff
@@ -1,13 +1,96 @@
-import { Component, ViewEncapsulation } from '@angular/core';
-
+import { Component } from '@angular/core';
+export class Hero {
+  id: number;
+  name: string;
+}
+const HEROES: Hero[] = [
+  { id: 11, name: 'Mr. Nice' },
+  { id: 12, name: 'Narco' },
+  { id: 13, name: 'Bombasto' },
+  { id: 14, name: 'Celeritas' },
+  { id: 15, name: 'Magneta' },
+  { id: 16, name: 'RubberMan' },
+  { id: 17, name: 'Dynama' },
+  { id: 18, name: 'Dr IQ' },
+  { id: 19, name: 'Magma' },
+  { id: 20, name: 'Tornado' }
+];
 @Component({
-  selector: 'toh-app',
-  encapsulation: ViewEncapsulation.None,
-  styles: [require('./app.component.scss')],
+  selector: 'my-app',
   template: `
-    <main>
-      <h1>Tour of Heroes</h1>
-    </main>
-  `
+    <h1>{{title}}</h1>
+    <h2>My Heroes</h2>
+    <ul class="heroes">
+      <li *ngFor="let hero of heroes"
+        [class.selected]="hero === selectedHero"
+        (click)="onSelect(hero)">
+        <span class="badge">{{hero.id}}</span> {{hero.name}}
+      </li>
+    </ul>
+    <div *ngIf="selectedHero">
+      <h2>{{selectedHero.name}} details!</h2>
+      <div><label>id: </label>{{selectedHero.id}}</div>
+      <div>
+        <label>name: </label>
+        <input [(ngModel)]="selectedHero.name" placeholder="name"/>
+      </div>
+    </div>
```

Comment:
HTML in JavaScript? Crazy idea, eh? But one that's picking up steam because of JSX and React. Some people will argue that HTML should be in a separate file. I used to feel this way as well. But now I embrace HTML in JavaScript. One of the main reasons for me...

If you put HTML in a separate file, it can become a tangled mass of shit-tons of HTML. But by keeping it in a JS file, too much HTML can start to look really ugly. Since ng2 is component-based by its very nature, if the HTML starts to become unsightly, it might be time to consider creating another component. Components are the shit. Focus should be on components.

Just bringing it up so after you are done with the TOH and begin to build your site, we can approach it with component architecture in mind.